### PR TITLE
REF: de-duplicate

### LINKED ISSUE #4801: REFACTOR: De-duplicate code
xref #4795

### PR DESCRIPTION
**Pull request by jbrockmendel**
Signed-off-by: Brock <jbrockmendel@gmail.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #4801
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by pyrito — APPROVED** [review]
Thank you for the cleanup!

**Comment by mvashishtha** [comment]
@jbrockmendel thanks for the cleanup. Could you please make issue(s) for the cleanup and reference them in the commit message to satisfy the commitlint?

**Comment by mvashishtha** [comment]
@jbrockmendel looking at [this failure](https://github.com/modin-project/modin/runs/7979742036?check_suite_focus=true) I think it's because [series shift](https://pandas.pydata.org/docs/reference/api/pandas.Series.shift.html) has `fill_value=None` and [dataframe shift](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.shift.html) has `fill_value=no_default`

File: modin/pandas/series.py [hunk 1]
```diff
@@ -803,12 +803,6 @@ def corr(self, other, method="pearson", min_periods=None):  # noqa: PR01, RT01,
             )
         )
 
-    def count(self, level=None):  # noqa: PR01, RT01, D200
```
Comment (mvashishtha) [inline modin/pandas/series.py:806]: this has extra params `numeric_only` and `axis`: https://github.com/modin-project/modin/runs/7982177596?check_suite_focus=true

File: modin/pandas/series.py [hunk 2]
```diff
@@ -911,12 +905,6 @@ def dot(self, other):  # noqa: PR01, RT01, D200
             query_compiler=self._query_compiler.dot(other, squeeze_self=True)
         )
 
-    def drop_duplicates(self, keep="first", inplace=False):  # noqa: PR01, RT01, D200
```
Comment (mvashishtha) [inline modin/pandas/series.py:914]: This has extra params `kwargs`: https://github.com/modin-project/modin/runs/7982177596?check_suite_focus=true

File: modin/pandas/series.py [hunk 3]
```diff
@@ -2137,12 +2119,6 @@ def ndim(self):  # noqa: RT01, D200
         """
         return 1
 
-    def nunique(self, dropna=True):  # noqa: PR01, RT01, D200
```
Comment (mvashishtha) [inline modin/pandas/series.py:2140]: This has extra params `axis`

**Review by mvashishtha — COMMENTED** [review]
@jbrockmendel there are some more series API failures. You can test with `pytest  modin/pandas/test/test_api.py::test_series_api_equality `

**Pull request closed without merging**